### PR TITLE
snap: improve error for snaps not available in the given context

### PR DESCRIPTION
**Pull request by mvo5**
When we have a snap in the store that is not available in the
channel that the user requested we currently show a "not-found"
error. But it is better to display more meaningful information.

The store provides a more meaningful error code now, so we use
it. This is a resurrect from #4443.

Fixes https://bugs.launchpad.net/snapstore/+bug/1665787

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: cmd/snap/error.go [hunk 1]
```diff
@@ -120,6 +120,11 @@ func errorToCmdMessage(snapName string, e error, opts *client.SnapOptions) (stri
 				msg = fmt.Sprintf(i18n.G("snap %%q not found (at least in channel %q)"), opts.Channel)
 			}
 		}
+	case client.ErrorKindRevisionNotAvailable:
+		// TRANSLATORS: %%q will become a %q for the snap name
```
Comment (bboozzoo) [inline cmd/snap/error.go]: There's no `%%q` in the format string.

File: daemon/api.go [hunk 2]
```diff
@@ -1193,6 +1187,18 @@ func (inst *snapInstruction) errToResponse(err error) Response {
 		default:
 			return InternalError("store.SnapNotFound with %d snaps", len(inst.Snaps))
 		}
+	case store.ErrRevisionNotAvailable:
+		switch len(inst.Snaps) {
+		case 1:
+			return SnapRevisionNotAvailable(inst.Snaps[0], err)
+		// store.ErrRevisionNotAvailable should only be
```
Comment (bboozzoo) [inline daemon/api.go]: I think this comment would make more sense when placed directly below `case store.ErrRevisionNotAvailable`.

File: tests/main/install-errors/task.yaml [hunk 3]
```diff
@@ -61,3 +61,21 @@ execute: |
...
     MATCH 'cannot install "ubuntu-core", please use "core" instead' < stderr.out
+
+    echo "============================================"
+
+    echo "Install a snap that is only available in the edge channel"
+    if snap install test-snapd-only-in-edge 2>stderr.out; then
+        echo "Installing test-snapd-only-in-edge should fail but did not"
+        exit 1
+    fi
+    MATCH 'error: snap not found in the given context.' < stderr.out
```
Comment (bboozzoo) [inline tests/main/install-errors/task.yaml]: The error message is still confusing. Shouldn't we suggest using `snap info <foo>` to see the details of the snap?

File: cmd/snap/error.go [hunk 4]
```diff
@@ -120,6 +120,11 @@ func errorToCmdMessage(snapName string, e error, opts *client.SnapOptions) (stri
...
+		msg = i18n.G(`
+snap not found in the given context. Please use "snap info %s" to list available releases.
```
Comment (chipaca) [inline cmd/snap/error.go]: We're using single quotes for quoting commands, for now at least

**Pull request merged**